plugins/woocommerce/src/Blocks/BlockTypes/AbstractInnerBlock.php

```diff
@@ -14,12 +14,53 @@ abstract class AbstractInnerBlock extends AbstractBlock {
 	protected $is_lazy_loaded = true;
 
 	/**
-	 * Registers the block type with WordPress using the metadata file.
+	 * Registers the block type with WordPress.
 	 *
-	 * The registration using metadata is now recommended. And it's required for "Inner Blocks" to
-	 * fix the issue of missing translations in the inspector (in the Editor mode)
+	 * This method determines whether to use compiled metadata or file-based metadata
+	 * for block registration.
 	 */
 	protected function register_block_type() {
```

Comment:
I wanted to do this, but ultimately couldn't because of a core bug. When using `register_block_type_from_metadata()` without a block.json file, if any styles are attached, the function fails to attach them. I reported it [here](https://core.trac.wordpress.org/ticket/61910).

This PR is a compromise which happens to work because [InnerBlocks never register styles](https://github.com/mreishus/woocommerce/blob/2ebbe685556920c86d21284711cae0f60713d5f6/plugins/woocommerce/src/Blocks/BlockTypes/AbstractInnerBlock.php#L62-L64), and the [outer blocks always do](https://github.com/mreishus/woocommerce/blob/2ebbe685556920c86d21284711cae0f60713d5f6/plugins/woocommerce/src/Blocks/BlockTypes/AbstractBlock.php#L346-L350).